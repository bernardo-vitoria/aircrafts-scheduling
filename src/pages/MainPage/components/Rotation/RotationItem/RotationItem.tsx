import React from 'react';
import { useRotationStack } from '../../../../../store';
import { 
  AirplaneTicketIcon, 
  ArrowForwardIcon, 
  Paper, 
  styled
} from "../../../../../components";
import { useTranslation } from 'react-i18next';
import "./RotationItem.css";
import { Flight } from '../../../types';

const Item = styled(Paper)(({ theme }) => ({
  ...theme.typography.body2,
  padding: theme.spacing(1),
  textAlign: "center",
  color: theme.palette.text.secondary,
  width: "50vh",
  marginTop: "8px",
  marginBottom: "8px",
  ":hover": { backgroundColor: "#FDFAF6" },
}));

type Props = {
  flight: Flight;
};

const RotationItem: React.FC<Props> = ({ flight }) => {
  const [t] = useTranslation();
  const {
    id,
    origin,
    destination,
    readable_arrival,
    readable_departure,
  } = flight;
  const { removeFlightFromRotation } = useRotationStack((state) => ({
    removeFlightFromRotation: state.removeFlightFromRotation,
  }));

  const handleOnClick = () => removeFlightFromRotation(flight);

  return (
    <Item onClick={handleOnClick}>
      <div className="card-header">
        <div className="card-title">
          {`${t("mainPage:rotations.flight")}: `}
          <b>{id}</b>
        </div>
        <div className="card-icon">
          <AirplaneTicketIcon color="primary" />
        </div>
      </div>
      <div className="card-body">
        <div className="card-body_column">
          <div style={{ textAlign: "left" }}>
            {`${t("mainPage:rotations.origin")}: `} <b>{origin}</b>
          </div>
          <div>
            {`${t("mainPage:rotations.departure")}: `}
            <b>{readable_departure}</b>
          </div>
        </div>
        <div className="card-body_column">
          <ArrowForwardIcon color="primary" sx={{ fontSize: 35 }} />
        </div>
        <div className="card-body_column">
          <div>
            {`${t("mainPage:rotations.destination")}: `} <b>{destination}</b>
          </div>
          <div style={{ textAlign: "right" }}>
            {`${t("mainPage:rotations.arrival")}: `} <b>{readable_arrival}</b>
          </div>
        </div>
      </div>
    </Item>
  );
};

export default RotationItem;