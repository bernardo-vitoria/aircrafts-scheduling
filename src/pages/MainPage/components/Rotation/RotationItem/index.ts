import RotationItem from "./RotationItem";

export { RotationItem };