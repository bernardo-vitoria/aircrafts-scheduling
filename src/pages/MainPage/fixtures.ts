const aircraft = { ident: "GABCD", type: "A320", economySeats: 186, base: "EGKK" };

const aircrafts = [
  { ident: "GABCD", type: "A320", economySeats: 186, base: "EGKK" },
];

const flight = {
  id: "AS1001",
  departuretime: 21600,
  arrivaltime: 26100,
  readable_departure: "06:00",
  readable_arrival: "07:15",
  origin: "LFSB",
  destination: "LFMN",
};

const flights = [
  {
    id: "AS1001",
    departuretime: 21600,
    arrivaltime: 26100,
    readable_departure: "06:00",
    readable_arrival: "07:15",
    origin: "LFSB",
    destination: "LFMN",
  },
  {
    id: "AS1002",
    departuretime: 27900,
    arrivaltime: 32100,
    readable_departure: "07:45",
    readable_arrival: "08:55",
    origin: "LFMN",
    destination: "LFSB",
  },
  {
    id: "AS1025",
    departuretime: 22800,
    arrivaltime: 28200,
    readable_departure: "06:20",
    readable_arrival: "07:50",
    origin: "LFSB",
    destination: "EDDH",
  },
  {
    id: "AS1026",
    departuretime: 30000,
    arrivaltime: 35100,
    readable_departure: "08:20",
    readable_arrival: "09:45",
    origin: "EDDH",
    destination: "LFSB",
  },
];

export {
    aircraft,
    aircrafts,
    flight,
    flights
};
