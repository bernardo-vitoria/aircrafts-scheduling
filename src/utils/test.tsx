// @ts-nocheck
import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { I18nextProvider } from "react-i18next";
import { QueryClient, QueryClientProvider } from "react-query";
import { MemoryRouter } from "react-router-dom"; // 4.0.0
import { queries } from "@testing-library/react";
import {
  render,
  RenderResult,
} from "@testing-library/react";
import i18n from "../config/i18nForTests";

const customRender = (ui: React.ReactElement, options = {}): RenderResult => {
  const CombinedProviders: React.FC = ({ children }) => {
    const config = {
      queries: {
        retry: false,
        cacheTime: 0,
        refetchOnWindowFocus: false,
        retryDelay: 0,
      },
    };

    const queryClient = new QueryClient({ defaultOptions: config });

    return (
        <QueryClientProvider client={queryClient}>
          <I18nextProvider i18n={i18n}>
            <MemoryRouter>{children}</MemoryRouter>
          </I18nextProvider>
        </QueryClientProvider>
    );
  };

  return render(ui, { wrapper: CombinedProviders, queries, ...options });
};

export {
  customRender as render
};