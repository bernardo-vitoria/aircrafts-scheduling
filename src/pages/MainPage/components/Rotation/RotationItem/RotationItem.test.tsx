import RotationItem from "./RotationItem";
import { render } from "../../../../../utils/test";
import { flight } from '../../../fixtures';

describe("RotationItem", () => {
  it("should match snapshot", () => {
    const { container } = render(<RotationItem flight={flight}/>);

    expect(container).toMatchSnapshot();
  });
});
