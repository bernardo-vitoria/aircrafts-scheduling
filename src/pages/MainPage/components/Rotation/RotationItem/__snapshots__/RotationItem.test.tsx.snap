// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RotationItem should match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-fqm90g-MuiPaper-root"
  >
    <div
      class="card-header"
    >
      <div
        class="card-title"
      >
        Flight: 
        <b>
          AS1001
        </b>
      </div>
      <div
        class="card-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
          data-testid="AirplaneTicketIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20.19 4H4c-1.1 0-1.99.9-1.99 2v4c1.1 0 1.99.9 1.99 2s-.89 2-2 2v4c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.81-2-1.81-2zm-2.46 9.3-8.86 2.36-1.66-2.88.93-.25 1.26.99 2.39-.64-2.4-4.16 1.4-.38 4.01 3.74 2.44-.65c.51-.14 1.04.17 1.18.68.13.51-.17 1.04-.69 1.19z"
          />
        </svg>
      </div>
    </div>
    <div
      class="card-body"
    >
      <div
        class="card-body_column"
      >
        <div
          style="text-align: left;"
        >
          Origin: 
           
          <b>
            LFSB
          </b>
        </div>
        <div>
          Departure: 
          <b>
            06:00
          </b>
        </div>
      </div>
      <div
        class="card-body_column"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-1ffxwdr-MuiSvgIcon-root"
          data-testid="ArrowForwardIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
          />
        </svg>
      </div>
      <div
        class="card-body_column"
      >
        <div>
          Destination: 
           
          <b>
            LFMN
          </b>
        </div>
        <div
          style="text-align: right;"
        >
          Arrival: 
           
          <b>
            07:15
          </b>
        </div>
      </div>
    </div>
  </div>
</div>
`;
